contrib/platform/src/com/sun/jna/platform/win32/VersionUtil.java

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2015 Michael Freeman, All Rights Reserved
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.  
+ */
+package com.sun.jna.platform.win32;
+
+import com.sun.jna.Memory;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.VerRsrc.VS_FIXEDFILEINFO;
+import com.sun.jna.platform.win32.Version;
+import com.sun.jna.platform.win32.Win32Exception;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.PointerByReference;
+
+/**
+ * Reads Windows Version info from files (the version details you can see by
+ * right-clicking and choosing properties)
+ * 
+ * @author mlfreeman[at]gmail.com
+ */
+public class VersionUtil {
+
+    /**
+     * Gets the file's version number info
+     * 
+     * @param filePath
+     *            The path to the file
+     * @return Either an array of {major, minor, build, revision} or null if no
+     *         version info could be obtained.
+     */
+    public static int[] getFileVersionNumbers(String filePath) {
+        IntByReference dwDummy = new IntByReference();
```

Comment:
Instead of returning an int[], return a class - e.g.:
```java
public class VersionInfo implements Serializable, Cloneable, Comparable<VersionInfo> {
    private static final long serialVersionUID = ...generate it via the IDE...;

    private int major = -1;
    private int minor = -1;
    private int buildNumber = -1;
    private int revision;

    public VersionInfo() {
          super();
    }

    public VersionInfo(int major, int minor, int buildNumber, int release) {
        this.minor/major/build/release = ...respective parameter...
    }

    public int getMajor/Minor/BuildNumber/Release() { 
        return ...respective member...
    }

    public void setMajor/Minor/BuildNumber/Release(int value) {
        this..minor/major/build/release = value;
    }

   @Override
    public int hashCode() {
        ...
    }

   @Override
    public boolean equals(Object o) {
        ...check if null, this == o and getClass() == o.getClass()...
       return compare((VersionInfo) o) == 0;
    }

    @Override
    public int compare(VersionInfo other) {
        ..check if other is null or this...
       ...compare major/minor/build/release - in this order...
    }

    @Override
    public VersionInfo clone() {
        try {
            return getClass().cast(super.clone());
        } catch(CloneNotSupportedException e) { // unexpected
            throw new RuntimeException("Failed to clone: " + toString(), e);
        }
    }

   @Override
    public String toString() {
        return getMajor() + "." + getMinor() + "." + getBuildNumber() + "." + getRevision();
    }

}
```
This way one does not need to remember which index in the array is the major/minor/build/revision...